File: CS/DocumentViewerWithToolbars/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using DevExpress.XtraReports.UI;
using DocumentViewerWithToolbars.Reports;
using System.Web.Mvc;

namespace DocumentViewerWithToolbars.Controllers
{
    public class HomeController : Controller
    {
        readonly XtraReport report = new MyXtraReport();

        public ActionResult Index()
        {
            return View(report);
        }

        public ActionResult DocumentViewerPartial()
        {
            return PartialView("_DocumentViewerPartial", report);
        }

        public ActionResult DocumentViewerPartialExport()
        {
            return DocumentViewerExtension.ExportTo(report, Request);
        }
    }
}

File: VB/DocumentViewerWithToolbars/Controllers/HomeController.vb
Imports DevExpress.Web.Mvc
Imports DevExpress.XtraReports.UI
Imports DocumentViewerWithToolbars.Reports
Imports System.Web.Mvc

Namespace DocumentViewerWithToolbars.Controllers
    Public Class HomeController
        Inherits Controller

        Private ReadOnly report As XtraReport = New MyXtraReport()

        Public Function Index() As ActionResult
            Return View(report)
        End Function

        Public Function DocumentViewerPartial() As ActionResult
            Return PartialView("_DocumentViewerPartial", report)
        End Function

        Public Function DocumentViewerPartialExport() As ActionResult
            Return DocumentViewerExtension.ExportTo(report, Request)
        End Function
    End Class
End Namespace
